### ISSUE TITLE
`unicorn/explicit-length-check` does not report when optional chaining

### ISSUE THREAD

**Issue by troywoy** [sindresorhus/eslint-plugin-unicorn#2252]
When an optional chain is involved, such as `data?.length`, the code is not correctly flagged by `unicorn/explicit-length-check`.

```tsx
// Does not error with "Use `.length > 0` when checking length is not zero.eslint(unicorn/explicit-length-check)"
if (data?.length) {
  ...
}
```

Issue was previously discussed here: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1263

codesandbox reproduction: https://codesandbox.io/p/devbox/h45722?file=%2Fsrc%2Findex.tsx%3A9%2C18

**Comment by fregante** [comment]
This is not a bug. You can't use `data?.length > 0` because `data?.length` can be `undefined`; `undefined > 0` requires type coercion. TypeScript would additionally mark it as an error.

**Comment by troywoy** [comment]
I tried to give a minimal reproduction, but the problem we encountered was in RJX with:
```tsx
{{something?.length && <SomeComponent /> }}
```
Which meant that if `something` existed it would render `0` instead of nothing. I don't know that I would call this "not a bug".

**Comment by fregante** [comment]
> in TSX

Let's say `something?.length` was flagged to be changed to `something?.length > 0`, now TypeScript complains and it's still a bug? What should be the solution here?

That particular issue isn't strictly related to `.length` because it applies to `items && <SomeComponent />`. See a broader discussion in:

https://github.com/jsx-eslint/eslint-plugin-react/issues/2073
